cvat/apps/dataset_manager/bindings.py

```diff
@@ -2151,11 +2154,67 @@ def reduce_fn(acc, v):
                 raise CvatImportError("Image {}: can't import annotation "
                     "#{} ({}): {}".format(item.id, idx, ann.type.name, e)) from e
 
-    for track in tracks.values():
+    finalize_tracks = {}
+    stop_frame = int(instance_data.meta[instance_data.META_FIELD]['stop_frame'])
+    for track_id, track in tracks.items():
+        finalize_tracks[track_id] = {
+            'label': track['label'],
+            'group': track['group'],
+            'source': track['source'],
+            'shapes': [],
+            'elements':{},
+        }
+        track['shapes'].sort(key=lambda t: t.frame)
+        prev_shape = track['shapes'][0]
+
+        for shape in track['shapes'][1:]:
+            if prev_shape.keyframe or prev_shape.outside:
+                finalize_tracks[track_id]['shapes'].append(prev_shape)
+            else:
+                has_skip = instance_data.frame_step < shape.frame - prev_shape.frame
+                if has_skip and not prev_shape.outside:
+                    prev_shape = prev_shape._replace(outside=True, keyframe=True,
+                            frame=prev_shape.frame + instance_data.frame_step)
+                    finalize_tracks[track_id]['shapes'].append(prev_shape)
+            prev_shape = shape
+
```

Comment:
I don't see this code adds keyframe = true to the 1st visible shape in each visible section of the track. Probably, it needs to be done somewhere (`if not prev_shape or prev_shape.outside:`).